### PR TITLE
chore: migrate to main branch

### LINKED ISSUE #110: Migrate to main branch
Contributing to googleapis/google-cloud-python#10579, switching the default branch to main branch. Also replacing all occurrences of master to main.

### PR DESCRIPTION
**Pull request by dandhlee**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-iam/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #110  🦕

### CODE REVIEW

**Review by plamut — APPROVED** [review]
LGTM, could not find any references to `master`, `whitelist`, etc. anymore apart from those in replacement rules.

BTW, when can we expect the code generator to generate the code using the `main` branch so that those replacement rules will not be needed anymore?

**Comment by dandhlee** [comment]
@plamut we'll sweep through the repositories first, then make template updates, then revert the replacement rules back. It shouldn't be long!

**Comment by plamut** [comment]
OK then, although it feels like extra work. :)

**Comment by dandhlee** [comment]
It is, but I think we're doing it so that we don't accidentally move stuff to the main branch and have things crash (template change sends 120+ PRs simultaneously), doing this slowly will ensure we don't break things along the way and minimize the impact as little as possible to prod, at the expense of some manhours which I think is affordable... maybe 😴

File: README.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ creation of service accounts, which you can use to authenticate to Google and ma
 - `Product Documentation`_
 
 .. |ga| image:: https://img.shields.io/badge/support-GA-gold.svg
-   :target: https://github.com/googleapis/google-cloud-python/blob/master/README.rst#general-availability
+   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#general-availability
```
Comment (parthea) [inline README.rst:13]: I get 404 with the new link

File: UPGRADING.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ In `google-cloud-iam<2.0.0`, parameters required by the API were positional para
 
 In the 2.0.0 release, all methods have a single positional parameter `request`. Method docstrings indicate whether a parameter is required or optional.
 
-Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation](https://github.com/googleapis/googleapis/blob/master/google/iam/credentials/v1/iamcredentials.proto#L49) specified by the API producer.
+Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation](https://github.com/googleapis/googleapis/blob/main/google/iam/credentials/v1/iamcredentials.proto#L49) specified by the API producer.
```
Comment (parthea) [inline UPGRADING.md:75]: I get 404 with the new link

File: samples/AUTHORING_GUIDE.md [hunk 3]
```diff
@@ -1 +1 @@
-See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md
\ No newline at end of file
+See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md
```
Comment (parthea) [inline samples/AUTHORING_GUIDE.md:1]: I get 404 with the new link

**Review by parthea — COMMENTED** [review]
Minor observations, otherwise LGTM.

File: README.rst (hunk 1 above)
Comment (plamut) [reply README.rst:13]: Presumably this will resolve itself once that repo is migrated and the  `main` branch is created?

Comment (dandhlee) [reply README.rst:13]: It will, but to Tony's point we shouldn't let users see a 404. I'll have it reverted for now.

Comment (plamut) [reply README.rst:13]: Ah, that's true, I thought the migration will happen simultaneously, but that's somewhat optimistic. :slightly_smiling_face:  Tony is right here.

File: UPGRADING.md (hunk 2 above)
Comment (dandhlee) [reply UPGRADING.md]: Done.

File: samples/AUTHORING_GUIDE.md (hunk 3 above)
Comment (dandhlee) [reply samples/AUTHORING_GUIDE.md]: Done.

**Pull request merged**